src/ol_infrastructure/applications/edxapp/__main__.py

```diff
@@ -1,28 +1,48 @@
 # TODO: Manage database object creation
+# TODO: Manage SES and export details for email management
+import base64
 import json
+from functools import partial
 from pathlib import Path
 from string import Template
 
 import pulumi_vault as vault
+import yaml
 from pulumi import Config, StackReference, export
-from pulumi_aws import ec2, get_caller_identity, iam, s3
+from pulumi_aws import acm, autoscaling, ec2, get_caller_identity, iam, lb, route53, s3
 from pulumi_consul import Node, Service, ServiceCheckArgs
 
-from bridge.lib.magic_numbers import DEFAULT_MYSQL_PORT, DEFAULT_REDIS_PORT
+from bridge.lib.magic_numbers import (
+    DEFAULT_HTTPS_PORT,
+    DEFAULT_MYSQL_PORT,
+    DEFAULT_REDIS_PORT,
+)
 from ol_infrastructure.components.aws.cache import OLAmazonCache, OLAmazonRedisConfig
 from ol_infrastructure.components.aws.database import OLAmazonDB, OLMariaDBConfig
 from ol_infrastructure.components.services.vault import (
     OLVaultDatabaseBackend,
     OLVaultMongoDatabaseConfig,
     OLVaultMysqlDatabaseConfig,
 )
-from ol_infrastructure.lib.aws.ec2_helper import default_egress_args
+from ol_infrastructure.lib.aws.ec2_helper import (
+    DiskTypes,
+    InstanceTypes,
+    default_egress_args,
+)
 from ol_infrastructure.lib.aws.iam_helper import IAM_POLICY_VERSION, lint_iam_policy
-from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.aws.route53_helper import acm_certificate_validation_records
+from ol_infrastructure.lib.ol_types import Apps, AWSBase
 from ol_infrastructure.lib.pulumi_helper import parse_stack
 from ol_infrastructure.lib.stack_defaults import defaults
 from ol_infrastructure.lib.vault import mysql_role_statements
 
+SSH_ACCESS_KEY_NAME = "salt-qa"
```

Comment:
Shouldn't this live in the stack definition?

Reply:
That was a placeholder. I am thinking about creating a new SSH key that is distinct from the Salt infrastructure.